framework/src/iotbus/iotbus_uart.c

```diff
@@ -176,6 +177,69 @@ int iotbus_uart_stop(iotbus_uart_context_h hnd)
 	return IOTBUS_ERROR_NONE;
 }
 
+int iotbus_uart_rxavailable(iotbus_uart_context_h hnd)
+{
+	int ret;
+	int fd;
+	struct _iotbus_uart_s *handle;
+
+	if (!hnd || !hnd->handle) {
+		return IOTBUS_ERROR_INVALID_PARAMETER;
+	}
+
+	handle = (struct _iotbus_uart_s *)hnd->handle;
+	fd = handle->fd;
+
+	ret = tccheckfifo(fd, TIOCS_AVAIL);
+	if (ret < 1) {
+		return IOTBUS_ERROR_QUEUE_EMPTY;
+	}
+
+	return IOTBUS_ERROR_NONE;
+}
+
+int iotbus_uart_txready(iotbus_uart_context_h hnd)
+{
+	int ret;
+	int fd;
+	struct _iotbus_uart_s *handle;
+
+	if (!hnd || !hnd->handle) {
+		return IOTBUS_ERROR_INVALID_PARAMETER;
+	}
+
+	handle = (struct _iotbus_uart_s *)hnd->handle;
+	fd = handle->fd;
+
+	ret = tccheckfifo(fd, TIOCS_READY);
+	if (ret < 1) {
+		return IOTBUS_ERROR_QUEUE_FULL;
+	}
+
+	return IOTBUS_ERROR_NONE;
+}
+
+int iotbus_uart_txempty(iotbus_uart_context_h hnd)
+{
+	int ret;
+	int fd;
+	struct _iotbus_uart_s *handle;
+
+	if (!hnd || !hnd->handle) {
+		return IOTBUS_ERROR_INVALID_PARAMETER;
+	}
+
+	handle = (struct _iotbus_uart_s *)hnd->handle;
+	fd = handle->fd;
+
+	ret = tccheckfifo(fd, TIOCS_EMPTY);
+	if (ret < 1) {
```

Comment:
I think it is better to check with `true`, not with `1`. Because `ioctl(fd, TIOCS_EMPTY, ,,,)` returns `true` for success case.
```
if (ret != true)
```